static/app/utils/url/safeURL.tsx

```diff
@@ -0,0 +1,24 @@
+/**
+ * Does not throw error on invalid input and returns the parsed URL object
+ * if the input is a valid URL, otherwise returns undefined.
+ * @param {string} input
+ * @param {string | undefined} base
+ * @returns {URL | undefined}
+ */
+export function safeURL(input: string | URL, base?: string | undefined): URL | undefined {
+  if ('canParse' in globalThis.URL) {
+    if (globalThis.URL.canParse(input, base)) {
+      return new URL(input, base);
+    }
+    return undefined;
+  }
```

Comment:
Having a try/catch is actually faster. If we look into the actual URL specification:

> Let parsedURL be the result of running the [API URL parser](https://url.spec.whatwg.org/#api-url-parser) on url with base, if given.

Ref: https://url.spec.whatwg.org/#dom-url-canparse

If we run `URL.canParse()` it basically executes the URL parser one time, without exceptions. And later, calling `new URL` does the same thing. Therefore, you're executing the URL parser twice in this example.

Reply:
True, though I always wondered though how much that actually hurts perf. Afaik try/catch is subject to deoptimization in V8? I'll move it back to just try/catch as I don't have anything to really argue with, double anything procedure for no reason are probably not good :)